actors/builtin/miner/miner_actor.go

```diff
@@ -1541,18 +1542,39 @@ func terminateSectors(rt Runtime, sectorNos *abi.BitField, terminationType power
 			rt.Abortf(exitcode.ErrIllegalState, "failed to expand faults")
 		}
 
+		var skippedSectorNos []uint64 // sectors we're not terminating because they either don't exist or have expired.
 		pledgeRequirementToRemove := abi.NewTokenAmount(0)
 		err = sectorNos.ForEach(func(sectorNo uint64) error {
 			sector, found, err := st.GetSector(store, abi.SectorNumber(sectorNo))
 			if err != nil {
 				return fmt.Errorf("failed to load sector %v: %w", sectorNo, err)
 			}
+
+			// If the sector wasn't found, skip termination. It may
+			// have already expired, have been terminated due to a
+			// faults, etc.
 			if !found {
-				rt.Abortf(exitcode.ErrNotFound, "no sector %v", sectorNo)
+				// XXX: Should we fail if the termination reason
+				// is "manual termination"? Or should we just
+				// let termination be idempotent?
+				skippedSectorNos = append(skippedSectorNos, sectorNo)
+				return nil
+			}
+
+			// If we're terminating the sector because it expired,
+			// make sure it has actually expired.
+			//
+			// We can hit this case if the sector was changed
+			// somehow but the termination wasn't removed from the
+			// SectorExpirations queue.
```

Comment:
I'm leery of this one. Is there a concrete case you know of? If it wasn't removed from the queue, it most likely wasn't added back to the queue somewhere else, so will never expire. We need the two representations of expiration to be consistent.

OTOH, I guess we could take advantage of this to avoid re-writing the queue entries to remove entries when we extend an expiration? We'll be paying an unnecessary loading of sector info here though.

Reply:
Sector number reuse? We don't currently (see TODOs) remove scheduled terminations when we terminate a sector early. If we then add a new sector with the same number, we could hit this case.